tests/runtime/variable

```diff
@@ -135,3 +135,35 @@ EXPECT (c, 0)
 NAME late map evaluation with ctx capture
 RUN {{BPFTRACE}} -e 'interval:ms:100 { $x = 1; for ($kv : @a) { print(($x, $kv.0, $kv.1)); exit(); } @a[1] = 1; }'
 EXPECT (1, 1, 1)
+
+NAME map declaration lruhash
+PROG let @a = lruhash(1); BEGIN { @a[1] = 1; @a[2] = 2; exit(); }
+EXPECT_REGEX_NONE .*WARNING: Map full; can't update element.*
+EXPECT @a[2]: 2
+
+NAME map declaration max entries
+PROG let @a = hash(1); BEGIN { @a[1] = 1; @a[2] = 2; exit(); }
+EXPECT_REGEX .*WARNING: Map full; can't update element.*
+EXPECT @a[1]: 1
+
+NAME map declaration default max entries
+PROG let @a = hash(); BEGIN { @a[1] = 1; @a[2] = 2; exit(); }
+EXPECT_REGEX_NONE .*WARNING: Map full; can't update element.*
+EXPECT @a[1]: 1
+EXPECT @a[2]: 2
+
+NAME map declaration percpuhash
+PROG let @a = percpuhash(1); BEGIN { @a[1] = count(); @a[2] = count(); exit(); }
+EXPECT_REGEX .*WARNING: Map full; can't update element.*
+EXPECT @a[1]: 1
+
+NAME map declaration percpuhash default max entries
+PROG let @a = percpuhash(); BEGIN { @a[1] = count(); @a[2] = count(); exit(); }
+EXPECT_REGEX_NONE .*WARNING: Map full; can't update element.*
```

Comment:
I still object to the functionality and the test highlights the reason :)

Now this test implicitly depends on the default size configuration, for example. We know in practice that it won't be less than 3, but this would theoretically bite us in exactly the same way it would bite a user -- they'll just find out their code is broken one day while they are trying to debug something (and they are lucky if they discover the problem quickly). No one should have to "discover" that there's a limit because that will never be a good experience -- it should be apparent and explicit up front